flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptive/CreatingExecutionGraphTest.java

```diff
@@ -146,6 +150,8 @@ public void testSuccessfulSlotAssignmentTransitionsToExecuting() throws Exceptio
             executionGraphWithvertexParallelismFuture.complete(
                     CreatingExecutionGraph.ExecutionGraphWithVertexParallelism.create(
                             executionGraph, new TestingVertexParallelism()));
+
+            assertThat(mockExecutionJobVertex.isExecutionDeployed(), is(true));
```

Comment:
There is no test which ensures the proper behavior if deploying fails in state `CreatingExecutionGraph`.